docs/developer-tools/ThrowHelper.md

```diff
@@ -0,0 +1,150 @@
+---
+title: ThrowHelper
+author: Sergio0694
+description: Helper methods to efficiently throw exceptions
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, debug, net core, net standard
+dev_langs:
+  - csharp
+---
+
+# ThrowHelper
+
+The [ThrowHelper](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.diagnostics.ThrowHelper) class is a helper type that can be used to efficiently throw exceptions. It is meant to support the [Guard](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.diagnostics.guard) APIs, and it should primarily be used in situations where developers need fine-grained controls over the exception types being thrown, or over the exact exception message to include.
+
+## How it works
+
+The `ThrowHelper` class exposes a series of `ThrowException` APIs that offer a 1:1 mapping to the constructors of all the available exception types. These methods are meant to be used in place of the classic `throw new Exception(...)` statement in code, as it will result in smaller and more efficient code. Put simply:
+
+```csharp
+// Replace this...
+throw new InvalidOperationException("Some custom message from my library");
+
+// ...with this
+ThrowHelper.ThrowInvalidOperationException("Some custom message from my library");
+```
+
+> [!NOTE]
+> This will also result in a slightly different stack trace, as the exception will be thrown by this helper method instead of directly by the original caller. The practical result is still the same as throwing an exception directly.
+
+The `Guard` APIs provide an easy to use abstraction over many common checks that might need to be performed, so it is recommended to first try to see if there is a `Guard` API that can be used, and only fallback on using `ThrowHelper` if that's not the case. As mentioned before, this might be if your library needs to throw a specific exception type not present in the `Guard` APIs, or if you need full control on the exact arguments being included in the exception being thrown.
+
+> [!NOTE]
```

Comment:
I don't think we need this note here, this is just restating what was in the introduction.

Reply:
Removed in https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/pull/370/commits/dc598defa183fbc19dca80e5a9d24f17bffafa0b.